src/wp-includes/block-supports/layout.php

```diff
@@ -41,45 +42,145 @@ function wp_register_layout_support( $block_type ) {
  * @param boolean $has_block_gap_support         Whether the theme has support for the block gap.
  * @param string  $gap_value                     The block gap value to apply.
  * @param boolean $should_skip_gap_serialization Whether to skip applying the user-defined value set in the editor.
- * @param string  $fallback_gap_value            The custom fallback value for block gap.
+ * @param string  $fallback_gap_value            The block gap value to apply.
+ * @param array   $block_spacing                 Custom spacing set on the block.
  * @return string CSS style.
  */
-function wp_get_layout_style( $selector, $layout, $has_block_gap_support = false, $gap_value = null, $should_skip_gap_serialization = false, $fallback_gap_value = '0.5em' ) {
-	$layout_type = isset( $layout['type'] ) ? $layout['type'] : 'default';
-
-	$style = '';
+function wp_get_layout_style( $selector, $layout, $has_block_gap_support = false, $gap_value = null, $should_skip_gap_serialization = false, $fallback_gap_value = '0.5em', $block_spacing = null ) {
+	$layout_type   = isset( $layout['type'] ) ? $layout['type'] : 'default';
+	$layout_styles = array();
 	if ( 'default' === $layout_type ) {
-		$content_size = isset( $layout['contentSize'] ) ? $layout['contentSize'] : '';
-		$wide_size    = isset( $layout['wideSize'] ) ? $layout['wideSize'] : '';
+		if ( $has_block_gap_support ) {
+			if ( is_array( $gap_value ) ) {
+				$gap_value = isset( $gap_value['top'] ) ? $gap_value['top'] : null;
+			}
+			if ( null !== $gap_value && ! $should_skip_gap_serialization ) {
+				array_push(
+					$layout_styles,
+					array(
+						'selector'     => "$selector > *",
+						'declarations' => array(
+							'margin-block-start' => '0',
+							'margin-block-end'   => '0',
+						),
+					),
+					array(
+						'selector'     => "$selector$selector > * + *",
+						'declarations' => array(
+							'margin-block-start' => $gap_value,
+							'margin-block-end'   => '0',
+						),
+					)
+				);
+			}
+		}
+	} elseif ( 'constrained' === $layout_type ) {
+		$content_size    = isset( $layout['contentSize'] ) ? $layout['contentSize'] : '';
+		$wide_size       = isset( $layout['wideSize'] ) ? $layout['wideSize'] : '';
+		$justify_content = isset( $layout['justifyContent'] ) ? $layout['justifyContent'] : 'center';
 
 		$all_max_width_value  = $content_size ? $content_size : $wide_size;
 		$wide_max_width_value = $wide_size ? $wide_size : $content_size;
 
 		// Make sure there is a single CSS rule, and all tags are stripped for security.
-		$all_max_width_value  = safecss_filter_attr( explode( ';', $all_max_width_value )[0] );
-		$wide_max_width_value = safecss_filter_attr( explode( ';', $wide_max_width_value )[0] );
+		// TODO: Use `safecss_filter_attr` instead - once https://core.trac.wordpress.org/ticket/46197 is patched.
```

Comment:
https://core.trac.wordpress.org/ticket/46197 has been closed as `fixed`. Is this `TODO` now actionable?

Reply:
Ah, this one shouldn't have been pulled in from Gutenberg as the core version was already using `safecss_filter_attr`. I'll look at updating Gutenberg to match. For now I've reverted these couple of lines to be as they were before 👍